navigation/navigation-runtime-ktx/src/main/java/androidx/navigation/NavController.kt

```diff
@@ -26,3 +31,19 @@ public inline fun NavController.createGraph(
     @IdRes startDestination: Int,
     builder: NavGraphBuilder.() -> Unit
 ): NavGraph = navigatorProvider.navigation(id, startDestination, builder)
+
+/**
+ * Creates and returns a [Flow] that will emit the currently active [NavBackStackEntry] whenever
+ * it changes. If there is no active [NavBackStackEntry], no item will be emitted.
+ */
+@Suppress("EXPERIMENTAL_API_USAGE")
+public fun NavController.getBackStackEntryFlow(): Flow<NavBackStackEntry> = callbackFlow {
```

Comment:
I'd probably suggest to shape it as extension property:
`public val NavController.backStackEntryFlow : .... `